src/app/configurations/allocations/calculations/calculations.service.js

```diff
@@ -63,52 +59,29 @@ angular.module('allocations')
       function setAllocations () {
         var view = 'allocations/custom-templates'
 
-        facilities.forEach(function (facility) {
-          facility.coverage = {}
-          facility.wastage = {}
-          facility.schedule = {}
-          facility.buffer = {}
-
-          // get custom template
-          return (function (v, key) {
-            var opts = {
-              include_docs: true,
-              key: key
-            }
-            return dbService.getView(v, opts)
-          }(view, facility._id))
-        })
-        return facilities
+        return dbService.getView(view, {include_docs: true})
+          .then(pouchUtil.pluckDocs)
+          .then(function (response) {
+            facilities.forEach(function (facility) {
+              facility.coverage = {}
+              facility.wastage = {}
+              facility.schedule = {}
+              facility.buffer = {}
+
+              for (var i in response) {
+                if (response[i]._id === (facility.name.trim().split(' ').join('-'))) {
+                  facility.custom_template = true
```

Comment:
i think fillTemplate and facility.custom_template = true should be called one time only for each facility with just one template.  as it is now, it is called for the amount custome templates we have.